packages/stockflux-launcher/src/App.js

```diff
@@ -1,69 +1,112 @@
-import React, { useRef, useEffect } from 'react';
 import cx from 'classnames';
+import { ScreenEdge, useDockWindow, useOptions } from 'openfin-react-hooks';
+import React, { useEffect, useState } from 'react';
+import Components from 'stockflux-components';
 import { OpenfinApiHelpers } from 'stockflux-core';
-import { useDockWindow, ScreenEdge } from 'openfin-react-hooks';
-import {
-  FaChevronUp,
-  FaChevronLeft,
-  FaChevronRight,
-  FaRegHandRock
-} from 'react-icons/fa';
-import AppShortcuts from './app-shortcuts/AppShortcuts';
+import './App.css';
 import FreeTextSearch from './free-text-search/FreeTextSearch';
+import getUndockedPosition from './helpers/getUndockedPosition';
+import { ReactComponent as LeftIcon } from './icons/left.svg';
+import { ReactComponent as RightIcon } from './icons/right.svg';
+import { ReactComponent as TopIcon } from './icons/top.svg';
+import Titlebar from './titlebar/Titlebar';
 import ToolBar from './toolbar/ToolBar';
-import CloseButton from './toolbar/CloseButton';
-import './App.css';
-
-const isLauncherHorizontal = edge => edge === ScreenEdge.TOP;
 
 export default () => {
+  const [options] = useOptions();
+  const defaultHeight = options ? options.defaultHeight : 75;
+  const defaultWidth = options ? options.defaultWidth : 50;
+  const isDockable = options ? options.customData.isDockable : false;
+  const [leftPosition, setLeft] = useState(0);
+  const [topPosition, setTop] = useState(0);
+  const [isHorizontal, setHorizontal] = useState(true);
```

Comment:
As I said a couple of reviews back, I think this is unnecessary. We already can check is it's Horizontal and this `useState` variable is just repeating the same logic. That info is stored in ScreenEdge and by checking it we can determine if it's horizontal or not.

Also, the name `isHorizontal` is less specific than checking screen edge and I think we should strive for specificity.

Instead, we can have a helper function called `isHorizontal` to check if the current position is `ScreenEdge.Top` or `ScreenEdge.None`.

Again, feel free to push back if you disagree.

Reply:
Discussed and raised #1351 to cover this as the edge/windowActions hook and dockedWidth is currently too tightly coupled so a state variable is necessary atm. 

Resolving anything `isHorizontal` related on this PR